Remove the custom Maybe type in test/pos/WhyLH.hs

`test/pos/WhyLH.hs` uses a type
```Haskell
data MaybeTy = Nothing | Just Ty
```
With the changes of ucsd-progsys/liquid-fixpoint#500 (and using the work in-progress branch) we can transition to
```Haskell
data Maybe a = Nothing | Just a
```
Unfortunately, we still can't use the `Maybe` coming from `liquid-base` because then LH complains
```
tests/pos/WhyLH.hs:119:17: error:
    • Illegal type specification for `Main.uappArgT`
    Main.uappArgT :: ctx:[Main.Ty] -> e:{e : Main.UExp | (if is$GHC.Maybe.Nothing (Main.inferType ctx e) then false else true)
                                                         && (if is$Main.UVar e then false else (if is$Main.ULam e then false else true))} -> {VV : () | (if is$GHC.Maybe.Nothing (Main.inferType ctx (Main.uapp2 e)) then GHC.Maybe.Nothing else (if is$GHC.Maybe.Nothing (Main.inferType ctx e) then GHC.Maybe.Nothing else GHC.Maybe.Just (Main.TyFun (lqdc##$select##GHC.Maybe.Just##1 (Main.inferType ctx (Main.uapp2 e))) (lqdc##$select##GHC.Maybe.Just##1 (Main.inferType ctx e))))) == Main.inferType ctx (Main.uapp1 e)}
    Sort Error in Refinement: {VV : Tuple | (if is$GHC.Maybe.Nothing (Main.inferType ctx (Main.uapp2 e)) then GHC.Maybe.Nothing else (if is$GHC.Maybe.Nothing (Main.inferType ctx e) then GHC.Maybe.Nothing else GHC.Maybe.Just (Main.TyFun (lqdc##$select##GHC.Maybe.Just##1 (Main.inferType ctx (Main.uapp2 e))) (lqdc##$select##GHC.Maybe.Just##1 (Main.inferType ctx e))))) == Main.inferType ctx (Main.uapp1 e)}
    Unbound symbol lqdc##$select##GHC.Maybe.Just##1 --- perhaps you meant: lqdc##$select##GHC.Tuple.(,)##1, lqdc##$select##GHC.Types.:##1, lqdc##$select##GHC.Tuple.(,,)##1 ?
    Just
    • 
    |
119 | {-@ uappArgT :: ctx:[Ty] -> e : { e:UExp | isUApp e && isJustTy (inferType ctx e) } -> { funTyM (inferType ctx (uapp2 e)) (inferType ctx e) == inferType ctx (uapp1 e) } @-}
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I have tried unsuccessfully to reproduce this in a smaller example. Probably it is best to address this after ucsd-progsys/liquid-fixpoint#500 is closed.

--- a/tests/pos/WhyLH.hs
+++ b/tests/pos/WhyLH.hs
@@ -7,7 +7,7 @@ module WhyLH where
 -- https://www.tweag.io/blog/2022-01-19-why-liquid-haskell/
 --
 import Language.Haskell.Liquid.ProofCombinators (pleUnfold)
-import Prelude hiding (Maybe(..), isJust, length, max)
+import Prelude hiding (length, max)
 
 {-@
 type Nat = {i:Int | 0 <= i}
@@ -79,13 +79,9 @@ length :: [a] -> Int
 length [] = 0
 length (_:xs) = 1 + length xs
 
--- XXX: Verification crashes when using @Maybe Ty@ instead of @MaybeTy@
-{-@ data MaybeTy = Nothing | Just Ty @-}
-data MaybeTy = Nothing | Just Ty
-
 {-@ reflect inferType @-}
-{-@ inferType :: ctx:[Ty] -> UExpN (length ctx) -> MaybeTy @-}
-inferType :: [Ty] -> UExp -> MaybeTy
+{-@ inferType :: ctx:[Ty] -> UExpN (length ctx) -> Maybe Ty @-}
+inferType :: [Ty] -> UExp -> Maybe Ty
 inferType ctx (UVar i) = Just (elemAt ctx i)
 inferType ctx (ULam t body) = case inferType (t:ctx) body of
   Just r -> Just (TyFun t r)
@@ -127,12 +123,13 @@ isUApp :: UExp -> Bool
 isUApp (UApp _ _) = True
 isUApp _ = False
 
+-- Either inline or both reflect and pleUnfold are needed by `uappArgT`.
 {-@ inline isJustTy @-}
-isJustTy :: MaybeTy -> Bool
+isJustTy :: Maybe a -> Bool
 isJustTy (Just _) = True
 isJustTy _ = False
 
 {-@ inline funTyM @-}
-funTyM :: MaybeTy -> MaybeTy -> MaybeTy
+funTyM :: Maybe Ty -> Maybe Ty -> Maybe Ty
 funTyM (Just a) (Just b) = Just (TyFun a b)
 funTyM _ _ = Nothing

